src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorIRLoweringPhase.cs

```diff
@@ -64,16 +54,31 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
 
             visitor.VisitBlock(syntaxTree.Root);
 
+            // In each lowering piece above, namespaces were tracked. We render them here to ensure every
+            // lowering action has a chance to add a source location to a namespace.
+
+            var i = builder.Current.Children.IndexOf(checksum) + 1;
+            foreach (var @namespace in namespaces)
+            {
+                var @using = new UsingStatementIRNode()
+                {
+                    Content = @namespace.Key,
+                    Source = @namespace.Value,
```

Comment:
When there are multiple identical usings, any chance one with null source appear later and overwrite one with source?